finality-aleph/src/network.rs

```diff
@@ -1,18 +1,53 @@
 use codec::{Decode, Encode};
 use futures::{channel::mpsc, stream::Stream, StreamExt};
 use parking_lot::Mutex;
-use rush::{NetworkCommand, NetworkEvent};
-use sc_network::{multiaddr, Event, ExHashT, NetworkService, PeerId, ReputationChange};
+use rush::{nodes::NodeIndex, Index, KeyBox as _};
+use sc_network::{multiaddr, Event, ExHashT, NetworkService, PeerId as ScPeerId, ReputationChange};
 use sp_runtime::traits::Block as BlockT;
-use std::{borrow::Cow, collections::HashMap, iter, marker::PhantomData, pin::Pin, sync::Arc};
+use std::{
+    borrow::Cow, collections::HashMap, hash::Hash, iter, marker::PhantomData, pin::Pin, sync::Arc,
+};
 
 use log::debug;
 
-use crate::{AuthorityId, Error, SessionId};
+use crate::{Error, Hasher, KeyBox, SessionId, Signature};
 
 #[cfg(test)]
 mod tests;
 
+#[derive(PartialEq, Eq, Clone, Debug, Hash)]
+struct PeerId(ScPeerId);
+
+impl From<PeerId> for ScPeerId {
+    fn from(wrapper: PeerId) -> Self {
+        wrapper.0
+    }
+}
+
+impl From<ScPeerId> for PeerId {
+    fn from(id: ScPeerId) -> Self {
+        PeerId(id)
+    }
+}
+
+impl Encode for PeerId {
+    // TODO: this probably has a constant number of bytes when encoded,
+    // so `size_hint` would be useful.
```

Comment:
Because it's undocumented and even if true most likely subject to change without notice. I guess I should remove the TODO then...

Reply:
Well, afterall it is just a hint, so even if the obvious meaning is that it is the size of encoding in bytes turns out wrong, it shouldn't break anything. I tend to implement it.

Reply:
Well, I would have to figure out how many bytes in a peer id and how these bytes translate into final size... I'm definitely too lazy to do all that, and even if I did this would be disgustingly unstable with changes in either `PeerId` or `Encode` of byte slices.